pkg/visibility/api/rest/pending_workload_CQ.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2023 The Kubernetes Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rest
+
+import (
+	"context"
+
+	"fmt"
+
+	"github.com/go-logr/logr"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/registry/rest"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	v1alpha1 "sigs.k8s.io/kueue/apis/api_extension/v1alpha1"
+	"sigs.k8s.io/kueue/pkg/queue"
+
+	_ "k8s.io/metrics/pkg/apis/metrics/install"
+)
+
+type pendingWorkloadsInCqREST struct {
+	kueueMgr *queue.Manager
+	log      logr.Logger
+}
+
+// var _ rest.KindProvider = &pendingWorkloadsInCqREST{}
+var _ rest.Storage = &pendingWorkloadsInCqREST{}
+var _ rest.GetterWithOptions = &pendingWorkloadsInCqREST{}
+var _ rest.Scoper = &pendingWorkloadsInCqREST{}
+var _ rest.SingularNameProvider = &pendingWorkloadsInCqREST{}
+
+func NewPendingWorkloadsInCqREST(kueueMgr *queue.Manager) *pendingWorkloadsInCqREST {
+	return &pendingWorkloadsInCqREST{
+		kueueMgr: kueueMgr,
+		log:      ctrl.Log.WithName("pending-workload-in-cq"),
+	}
+}
+
+// New implements rest.Storage interface
+func (m *pendingWorkloadsInCqREST) New() runtime.Object {
+	return &v1alpha1.PendingWorkloadSummary{}
+}
+
+// Destroy implements rest.Storage interface
+func (m *pendingWorkloadsInCqREST) Destroy() {
+}
+
+// Kind implements rest.KindProvider interface
+// func (m *pendingWorkloadsInCqREST) Kind() string {
+// return "PendingWorkloadsInClusterQueue"
+// }
+
+// Get implements rest.Getter interface
+// It fetches information about pending workloads and returns according to query params
+func (m *pendingWorkloadsInCqREST) Get(ctx context.Context, name string, opts runtime.Object) (runtime.Object, error) {
+	pendingWorkloadOpts, ok := opts.(*v1alpha1.PendingWorkloadOptions)
+	if !ok {
+		return nil, fmt.Errorf("invalid options object: %#v", opts)
+	}
+	limit := pendingWorkloadOpts.Limit
+	offset := pendingWorkloadOpts.Offset
+
+	var wls []v1alpha1.PendingWorkloadSummary
+	for pos, val := range m.kueueMgr.GetPendingWorkloadsInfo(name) {
+		if pos > int(offset) && pos <= int(offset)+int(limit) {
```

Comment:
```suggestion
		if pos >= int(offset) && pos <= int(offset)+int(limit) {
```
when offset=0, we don't want to skip any pending workloads, but return all.

Reply:
Fixed